modules/core/shared/src/main/scala/coursier/core/Reconciliation.scala

```diff
@@ -0,0 +1,118 @@
+package coursier.core
+
+/**
+ * Represents a reconciliation strategy given a dependency conflict.
+ */
+abstract class Reconciliation {
+  /**
+   * Reconcile multiple version candidate.
+   *
+   * Returns `None` in case of conflict.
+   */
+  def reconcileVersions(module: Module, versions: Seq[String]): Option[String]
+}
+
+object Reconciliation {
+  private[coursier] final val LatestIntegration = "latest.integration"
+  private[coursier] final val LatestRelease = "latest.release"
+  private[coursier] final val LatestStable = "latest.stable"
+
+  abstract class AbstractReconciliation extends Reconciliation {
+    protected def mergeVersionConstraints(constraints: Seq[VersionConstraint]): Option[VersionConstraint]
+
+    /**
+     * Reconcile multiple version candidate.
+     *
+     * Returns `None` in case of conflict.
+     */
+    override def reconcileVersions(module: Module, versions: Seq[String]): Option[String] = {
```

Comment:
I'm not sure this is worth sharing across implementations… `RelaxedReconciliation` actually needs a slightly different processing here: if `retainedLatestOpt` is non empty, it should be retained (and the intevals / specific versions discarded). Else it should look at other stuff. The goal of retaining both a `latest.*` and an interval in the default `Reconciliation` is to allow to check that the version that get selected by `latest.*` is indeed in the interval (it's handled from `ResolutionProcess` IIRC). We don't need that with relaxed reconciliation.

Reply:
Addressed in https://github.com/coursier/coursier/pull/1293/commits/63d43919bc3a5b940bb93297e590588d02d75819